README.md

```diff
@@ -29,15 +29,15 @@ Add this dependency to your project's POM:
 <dependency>
   <groupId>com.stripe</groupId>
   <artifactId>stripe-java</artifactId>
-  <version>22.30.0</version>
+  <version>23.0.0</version>
```

Comment:
We're doing the version change in this PR too?

Reply:
Java tests have a few places that check versions in readme and changelog, wanted to make it happy. Should be fine to merge.